src/bindings/python/src/pyopenvino/graph/shape.cpp

```diff
@@ -44,9 +44,28 @@ void regclass_graph_Shape(py::module m) {
         if (key < 0) {
             key += v.size();
         }
-        return v[key];
+        if (key >= 0 && key < static_cast<int64_t>(v.size())) {
+            return v[key];
+        }
+        throw py::index_error("Index out of range");
     });
 
+    shape.def("__eq__", [](const ov::Shape& self, const py::list& lst) {
+            if (self.size() != lst.size()) return false;
```

Comment:
Here the same. I think that common templated function may be created for this logic.